### PR TITLE
Frontend bugfix + added some tests for helper functions

### LINKED ISSUE #101: måske mere input validation for ingredients - du kan fx skrive emojis, og det skal vel kun være bogstaver og måske tal og parenteser

### LINKED ISSUE #195: yeet engelsk tekst fra websiden (fx no options i react-selects)

### LINKED ISSUE #221: add toastcontainer to fullRecipeViewPage

### PR DESCRIPTION
**Pull request by Lildhansen**
Fixes #101 
Fixes #221 
Fixes #195 

now when you try to create user, toasts appear for all wrong inputs

also added tests for IngredientsHelper and InputValidation

also fixed the additional ingredients modal such that the modal now updates with the ingredients you selected before

### CODE REVIEW

**Comment by NikolajKrogh** [comment]
Toast burde komme i omvendt rækkefølge end de gør nu
image

File: P7Internet.Frontend/react-app/src/__tests__/helperFunctionsTests/ingredientsHelperTest.js [hunk 1]
```diff
@@ -1,6 +1,10 @@
 import { convertIngredientsToIngredientObjects } from "../../helperFunctions/ingredientHelper";
+import { ingredientIsOwned } from "../../helperFunctions/ingredientHelper";
+
 import "@testing-library/jest-dom";
 
+//convertIngredientsToIngredientObjects
```
Comment (NikolajKrogh) [inline P7Internet.Frontend/react-app/src/__tests__/helperFunctionsTests/ingredientsHelperTest.js]: Er det ikke det samme som testen allerede siger

File: P7Internet.Frontend/react-app/src/__tests__/helperFunctionsTests/ingredientsHelperTest.js [hunk 2]
```diff
@@ -18,3 +22,24 @@ test("ingredients can be converted to ingredient objects", () => {
     id: expect.anything(),
   });
 });
+
+// ingredientIsOwned
```
Comment (NikolajKrogh) [inline P7Internet.Frontend/react-app/src/__tests__/helperFunctionsTests/ingredientsHelperTest.js]: Samme igen

File: P7Internet.Frontend/react-app/src/__tests__/helperFunctionsTests/inputValidationTest.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
+import { screen, render } from "@testing-library/react";
+import {
+  userInputValidation,
+  ingredientInputValidation,
+} from "../../helperFunctions/inputValidation";
+import "@testing-library/jest-dom";
+import { ToastContainer } from "react-toastify";
+
+//userInputValidation
```
Comment (NikolajKrogh) [inline P7Internet.Frontend/react-app/src/__tests__/helperFunctionsTests/inputValidationTest.js]: Det samme som står i testen

File: P7Internet.Frontend/react-app/src/__tests__/helperFunctionsTests/inputValidationTest.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    )
+  ).toBeInTheDocument();
+  expect(
+    await screen.findByText(
+      /Kodeordet skal bestå af mindst et tal, et stort bogstav, et lille bogstav og være mellem 6 og 20 tegn langt uden brug af specielle tegn./
+    )
+  ).toBeInTheDocument();
+});
+
+//ingredientInputValidation
```
Comment (NikolajKrogh) [inline P7Internet.Frontend/react-app/src/__tests__/helperFunctionsTests/inputValidationTest.js]: Samme igen

**Review by NikolajKrogh — APPROVED** [review]

**Review by sebxGit — APPROVED** [review]

**Pull request merged**